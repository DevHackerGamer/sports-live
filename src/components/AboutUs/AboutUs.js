// components/AboutUs/AboutUs.js
import React from 'react';

const AboutUs = ({ setShowAboutUs, setActiveTab }) => {
  return (
    <div className="about-us-container">
      <div className="container">
        <h2>About Sports Live</h2>
        <div className="about-content">
          <p>Sports Live is the premier destination for football enthusiasts who want real-time updates, comprehensive statistics, and in-depth coverage of all major football leagues around the world.</p>
          <h3>Our Mission</h3>
          <p>To provide football fans with the most accurate, timely, and comprehensive football data and match coverage in an intuitive and engaging platform.</p>
          <h3>What We Offer</h3>
          <ul>
            <li>Real-time match updates and live scores</li>
            <li>Detailed player and team statistics</li>
            <li>League standings and tournament progress</li>
            <li>Personalized favorites system to follow your preferred teams</li>
            <li>Comprehensive coverage of Premier League, La Liga, Serie A, Bundesliga, Ligue 1, and European competitions</li>
          </ul>
          <h3>Our Team</h3>
          <p>We are a passionate group of football enthusiasts, data analysts, and developers dedicated to creating the best football experience for fans worldwide.</p>

          <h3>Contact Us</h3>
          <p>Have questions or feedback? Reach out to us at support@sportslive.com</p>
        </div>
        <button 
          className="btn btn-secondary mt-3" 
          onClick={() => {
            setShowAboutUs(false);
            setActiveTab('home');
          }}
        >
          Back to Dashboard
        </button>
      </div>
    </div>
  );
};

export default AboutUs;