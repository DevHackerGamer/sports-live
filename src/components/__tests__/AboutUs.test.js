import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import AboutUs from '../AboutUs/AboutUs';

describe('AboutUs Component', () => {
  const mockSetShowAboutUs = jest.fn();
  const mockSetActiveTab = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders all AboutUs content correctly', () => {
    render(
      <AboutUs 
        setShowAboutUs={mockSetShowAboutUs}
        setActiveTab={mockSetActiveTab}
      />
    );

    // Check main heading
    expect(screen.getByText('About Sports Live')).toBeInTheDocument();

    // Check main description
    expect(screen.getByText(/Sports Live is the premier destination for football enthusiasts/i)).toBeInTheDocument();

    // Check section headings
    expect(screen.getByText('Our Mission')).toBeInTheDocument();
    expect(screen.getByText('What We Offer')).toBeInTheDocument();
    expect(screen.getByText('Our Team')).toBeInTheDocument();
    expect(screen.getByText('Contact Us')).toBeInTheDocument();

    // Check mission statement
    expect(screen.getByText(/To provide football fans with the most accurate/i)).toBeInTheDocument();

    // Check features list
    expect(screen.getByText('Real-time match updates and live scores')).toBeInTheDocument();
    expect(screen.getByText('Detailed player and team statistics')).toBeInTheDocument();
    expect(screen.getByText('League standings and tournament progress')).toBeInTheDocument();
    expect(screen.getByText('Personalized favorites system to follow your preferred teams')).toBeInTheDocument();
    expect(screen.getByText('Comprehensive coverage of Premier League, La Liga, Serie A, Bundesliga, Ligue 1, and European competitions')).toBeInTheDocument();

    // Check team description
    expect(screen.getByText(/We are a passionate group of football enthusiasts/i)).toBeInTheDocument();

    // Check contact information
    expect(screen.getByText(/support@sportslive.com/i)).toBeInTheDocument();

    // Check back button
    expect(screen.getByText('Back to Dashboard')).toBeInTheDocument();
  });

  it('calls setShowAboutUs and setActiveTab when back button is clicked', () => {
    render(
      <AboutUs 
        setShowAboutUs={mockSetShowAboutUs}
        setActiveTab={mockSetActiveTab}
      />
    );

    const backButton = screen.getByText('Back to Dashboard');
    fireEvent.click(backButton);

    expect(mockSetShowAboutUs).toHaveBeenCalledWith(false);
    expect(mockSetActiveTab).toHaveBeenCalledWith('home');
  });

  it('renders with correct structure and classes', () => {
    const { container } = render(
      <AboutUs 
        setShowAboutUs={mockSetShowAboutUs}
        setActiveTab={mockSetActiveTab}
      />
    );

    // Check container classes
    const aboutContainer = container.querySelector('.about-us-container');
    expect(aboutContainer).toBeInTheDocument();

    const innerContainer = container.querySelector('.container');
    expect(innerContainer).toBeInTheDocument();

    const aboutContent = container.querySelector('.about-content');
    expect(aboutContent).toBeInTheDocument();

    // Check button classes
    const backButton = screen.getByText('Back to Dashboard');
    expect(backButton).toHaveClass('btn', 'btn-secondary', 'mt-3');
  });

  it('renders all list items in the features section', () => {
    render(
      <AboutUs 
        setShowAboutUs={mockSetShowAboutUs}
        setActiveTab={mockSetActiveTab}
      />
    );

    const listItems = screen.getAllByRole('listitem');
    expect(listItems).toHaveLength(5);

    // Verify specific list items
    expect(listItems[0]).toHaveTextContent('Real-time match updates and live scores');
    expect(listItems[1]).toHaveTextContent('Detailed player and team statistics');
    expect(listItems[2]).toHaveTextContent('League standings and tournament progress');
    expect(listItems[3]).toHaveTextContent('Personalized favorites system to follow your preferred teams');
    expect(listItems[4]).toHaveTextContent('Comprehensive coverage of Premier League, La Liga, Serie A, Bundesliga, Ligue 1, and European competitions');
  });

  it('matches snapshot', () => {
    const { container } = render(
      <AboutUs 
        setShowAboutUs={mockSetShowAboutUs}
        setActiveTab={mockSetActiveTab}
      />
    );

    expect(container).toMatchSnapshot();
  });
});