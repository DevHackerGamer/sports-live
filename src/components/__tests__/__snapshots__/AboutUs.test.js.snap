// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutUs Component matches snapshot 1`] = `
<div>
  <div
    class="about-us-container"
  >
    <div
      class="container"
    >
      <h2>
        About Sports Live
      </h2>
      <div
        class="about-content"
      >
        <p>
          Sports Live is the premier destination for football enthusiasts who want real-time updates, comprehensive statistics, and in-depth coverage of all major football leagues around the world.
        </p>
        <h3>
          Our Mission
        </h3>
        <p>
          To provide football fans with the most accurate, timely, and comprehensive football data and match coverage in an intuitive and engaging platform.
        </p>
        <h3>
          What We Offer
        </h3>
        <ul>
          <li>
            Real-time match updates and live scores
          </li>
          <li>
            Detailed player and team statistics
          </li>
          <li>
            League standings and tournament progress
          </li>
          <li>
            Personalized favorites system to follow your preferred teams
          </li>
          <li>
            Comprehensive coverage of Premier League, La Liga, Serie A, Bundesliga, Ligue 1, and European competitions
          </li>
        </ul>
        <h3>
          Our Team
        </h3>
        <p>
          We are a passionate group of football enthusiasts, data analysts, and developers dedicated to creating the best football experience for fans worldwide.
        </p>
        <h3>
          Contact Us
        </h3>
        <p>
          Have questions or feedback? Reach out to us at support@sportslive.com
        </p>
      </div>
      <button
        class="btn btn-secondary mt-3"
      >
        Back to Dashboard
      </button>
    </div>
  </div>
</div>
`;
